### PR TITLE
{AKS} Refactor 4th part of the update sub-command in acs module

### PR DESCRIPTION
**Pull request by FumingZhang**
**Description**

- refactor uptime_sla & no_uptime_sla.
- fix typos.

Unit Test Results:

- "covered_lines": 1297,
- "num_statements": 1320,
- "percent_covered": 98.25757575757575,
- "missing_lines": 23,
- "excluded_lines": 0

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py [hunk 1]
```diff
@@ -3301,22 +3301,64 @@ def get_cluster_autoscaler_profile(self) -> Union[Dict[str, str], None]:
...
         uptime_sla = self.raw_param.get("uptime_sla")
-        # try to read the property value corresponding to the parameter from the `mc` object
-        if (
-            self.mc and
-            self.mc.sku and
-            self.mc.sku.tier is not None
-        ):
-            uptime_sla = self.mc.sku.tier == "Paid"
+        # In create mode, try to read the property value corresponding to the parameter from the `mc` object
+        if self.decorator_mode == DecoratorMode.CREATE:
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/decorator.py:3312]: do we have code path if the decorator_mode is other value?

Comment (andyliuliming) [reply src/azure-cli/azure/cli/command_modules/acs/decorator.py:3312]: I mean. "blabla"

Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/decorator.py:3318]: Good catch, changed in commit [#d337899](https://github.com/Azure/azure-cli/pull/19699/commits/d33789913acfe271b036bd585ecf1d0049d72635).

**Review by andyliuliming — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**